ui/src/app/job-details/resources/resources.component.ts

```diff
@@ -18,78 +17,64 @@ import {GcsService} from '../../core/gcs.service';
   templateUrl: './resources.component.html',
   styleUrls: ['./resources.component.css'],
 })
-export class JobResourcesComponent implements OnChanges {
+export class JobResourcesComponent implements OnInit {
   @Input() job: JobMetadataResponse;
 
-  script: string = '';
+  sourceFile: string = '';
   inputs: Array<string> = [];
-  logFileData: Map<string, string> = new Map();
-  logFiles: Array<string> = [];
   outputs: Array<string> = [];
-  currentResourcesTab: string = '';
+  logFileData: Map<string, string> = new Map();
 
-  constructor(private readonly gcsService: GcsService) {}
+  // Map of tab "id" to tab title. This prevents collisions if there was, for
+  // example, a log file named "inputs".
+  tabTitles: Map<string, string> = new Map();
+  currentTab: string = '';
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.job = changes.job.currentValue;
-    if (this.job.extensions) {
-      if (this.job.extensions.logs) {
-        this.logFiles = Object.keys(this.job.extensions.logs).sort();
-        for (let file of Object.keys(this.job.extensions.logs)) {
-          let bucket = ResourceUtils.getResourceBucket(this.job.extensions.logs[file]);
-          let object = ResourceUtils.getResourceObject(this.job.extensions.logs[file]);
-          this.gcsService.readObject(bucket, object).then(data => {
-            if (data.length > 0) {
-              this.logFileData.set(file, data);
-            } else {
-              // Remove any files from the tab which are completely empty
-              this.logFiles.splice(this.logFiles.indexOf(file), 1);
-            }
-          })
-        }
-      }
-
-      if (this.job.extensions.script) {
-        this.script = this.job.extensions.script;
-      }
-    }
+  constructor(private readonly gcsService: GcsService) {}
 
+  ngOnInit() {
     if (this.job.inputs) {
       this.inputs = Object.keys(this.job.inputs).sort();
+      this.tabTitles.set('Inputs', 'inputs');
```

Comment:
Isn't this going in the wrong direction?

Reply:
(see other comment on the tabId approach)

Reply:
Fixed with updated strategy.